server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java

```diff
@@ -51,12 +53,14 @@
  * are copied, but a selected few. It is possible to control what headers are copied over by returning them in
  * {@link ActionPlugin#getRestHeaders()}.
  */
-public abstract class BaseRestHandler extends AbstractComponent implements RestHandler {
+public abstract class BaseRestHandler implements RestHandler {
 
     public static final Setting<Boolean> MULTI_ALLOW_EXPLICIT_INDEX =
         Setting.boolSetting("rest.action.multi.allow_explicit_index", true, Property.NodeScope);
 
     private final LongAdder usageCount = new LongAdder();
+    @Deprecated
+    protected static Logger logger = LogManager.getLogger(BaseRestHandler.class);
```

Comment:
why is this logger deprecated?

Reply:
should this be for getClass() instead of BaseRestHandler.class?

Reply:
Yeah, `getClass()` I think.

Reply:
Hi, Thanks for the comments. I am still understanding how the project works. Anyway, to use `getClass()` the logger cannot be static. Is that ok? (Or am I seeing something wrong here)

Reply:
Hi, Thanks for the comments. I am still understanding how the project works. Anyway, to use `getClass()` the logger cannot be static. Is that ok? (Or am I seeing something wrong here)